window/src/lib.rs

```diff
@@ -175,6 +176,12 @@ pub enum WindowEvent {
     AppearanceChanged(Appearance),
 
     Notification(Box<dyn Any + Send + Sync>),
+
+    // A file is being dragged into the window
+    DraggedFile(PathBuf),
+
+    // A file is been dropped into the window
+    DroppedFile(PathBuf),
```

Comment:
I think these should be `Vec` to model the multiple file vs. single file case so that we can make some smarter decisions about how to propagate drops to the application in the terminal; see my other comments about escaping, spaces and multiple files!

```suggestion
    // A file is being dragged into the window
    DraggedFile(Vec<PathBuf>),

    // A file is been dropped into the window
    DroppedFile(Vec<PathBuf>),
```

Reply:
@Junnplus What will happen if we drag/drop a dir here? Do we need to take that case into consideration?

Reply:
Dir are the same as files, just a path.